src/Elastic.Apm.Extensions.Hosting/HostBuilderExtensions.cs

```diff
@@ -57,15 +56,22 @@ public static IHostBuilder UseElasticApm(this IHostBuilder builder, params IDiag
 				{
 					if (Agent.IsConfigured) return Agent.Instance;
 
-					var apmAgent = new ApmAgent(sp.GetService<AgentComponents>());
+					var newAgentInstance = new ApmAgent(sp.GetService<AgentComponents>());
 					Agent.Setup(sp.GetService<AgentComponents>());
-					return apmAgent;
+					return newAgentInstance;
```

Comment:
can agent components be resolved once, and passed to both agent instance and `Agent.Setup(...)` i.e.

```c#
var components = sp.GetService<AgentComponents>();
var newAgentInstance = new ApmAgent(components );
Agent.Setup(components);					
return newAgentInstance;
```

Also, does this mean there are potentially two instances of an agent in the application - a singleton instance created by this method, and the lazy instance on `Agent` that's used internally?

Reply:
> Also, does this mean there are potentially two instances of an agent in the application - a singleton instance created by this method, and the lazy instance on Agent that's used internally?

Hahh - that's a good point - we should not new up an `ApmAgent` instance here. Technically they were 2 instances, but they were sharing the `AgentComponents`, but still - best is to just return `Agent.Instance`. I changed this part.

> can agent components be resolved once, and passed to both agent instance and Agent.Setup(...) i.e.

With the previous change this is not needed. Otherwise I'd agree.